index.js

```diff
@@ -85,11 +85,12 @@ export function mergeTrees(matchingTrees, context) {
 
 class Rule {
 
-    constructor(name, parent, style, filter) {
+    constructor(name, parent, style, filter, properties) {
```

Comment:
Minor note: seems like we have `whiteList` which partially centralizes the handling of object properties we want to preserve, but then a couple other places where we need to specify new property name explicitly. It would be possible to consolidate this all and make it controllable just by changing `whiteList`... on the other hand, it might be more opaque as to what was going on, and adding any additional white-listed properties like this is probably going to be a pretty uncommon occurrence... so I figured I would just leave it like this.